### PR TITLE
Fix Unlock for read-lock usecase

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-core/src/main/java/water/Lockable.java [hunk 1]
```diff
@@ -212,21 +212,25 @@ public T unlock( Key<Job> job_key, boolean exact ) {
...
     Unlock( Key<Job> job_key, boolean exact ) { _job_key = job_key; _exact = exact;}
-    @Override public Lockable atomic(Lockable old) {
+    @Override public Lockable<T> atomic(Lockable<T> old) {
       assert !_exact || old != null : "Trying to unlock null!";
       assert !_exact || old.is_locked(_job_key) : "Can't unlock: Not locked!";
-      if( _exact || old.is_locked(_job_key) )
-        set_unlocked(old._lockers,_job_key);
-      return Lockable.this;
+      final Lockable<T> l = old == null || old.is_wlocked() ?
+              Lockable.this : old;
```
Comment (sebhrusen) [inline h2o-core/src/main/java/water/Lockable.java:228]: looks good this way

**Review by sebhrusen — COMMENTED** [review]
like it!
Will fully approve once you're able to remove the temp jenkins changes.

**Comment by michalkurka** [comment]
Thank you @sebhrusen. I am going to remove the temp changes once the test pass.

**Comment by michalkurka** [comment]
@sebhrusen temporary changes in build scripts removed, please take another look

**Review by sebhrusen — APPROVED** [review]

**Comment by michalkurka** [comment]
thank you for the review and discussion

**Pull request merged**